spread-tests/main/discard-ns/task.yaml

```diff
@@ -18,4 +18,3 @@ restore: |
     umount /run/snapd/ns
     rm /run/snapd/ns/foo.mnt
     rm /run/snapd/ns/foo.lock
-    rmdir /run/snapd/ns
```

Comment:
What about "rmdir /run/snapd/ns || true" instead (with a comment that it is ok when it can't be removed). This way we get the cleanup if it can be done.